src/compiler/types.ts

```diff
@@ -2651,8 +2662,13 @@ namespace ts {
         getTypeAtLocation(node: Node): Type;
         getTypeFromTypeNode(node: TypeNode): Type;
         signatureToString(signature: Signature, enclosingDeclaration?: Node, flags?: TypeFormatFlags, kind?: SignatureKind): string;
+        /* @internal */ signatureToString(signature: Signature, enclosingDeclaration?: Node, flags?: TypeFormatFlags, kind?: SignatureKind, writer?: EmitTextWriter): string;

```

Comment:
Rather than overloading these, why not have:

```ts
  /* @internal */ writeSignature(writer: EmitTextWriter, signature: Signature, ...): void;
  /* @internal */ writeType(writer: EmitTextWriter, type: Type, ...): void;
  /* @internal */ writeSymbol(writer: EmitTextWriter, symbol: Symbol, ...): void;
  /* @internal */ writeTypePredicate(writer: EmitTextWriter, predicate: TypePredicate, ...): void;
```

That is similar to how `Printer` splits up internal and public members.